lib/index.js

```diff
@@ -35,42 +39,66 @@ exports.register = function (server, options, next) {
   });
 
   var baseModel;
+  var baseCollection;
   if (options.base) {
-    baseModel = options.base(bookshelf);
-  } else {
+    if (typeof options.base === 'function') {
+      baseModel = options.base(bookshelf);
+    } else {
+      if (options.base.model) {
+        baseModel = options.base.model(bookshelf);
+      }
+      if (options.base.collection) {
+        baseCollection = options.base.collection(bookshelf);
+      }
+    }
+  }
+
+  if (!baseModel) {
     baseModel = bookshelf.Model.extend({});
   }
+  if (!baseCollection) {
+    baseCollection = bookshelf.Collection.extend({});
+  }
 
-  var gFiles = [];
-  for (var i = options.models.length - 1; i >= 0; i--) {
-    var filename = options.models[i];
-    if (!glob.hasMagic(filename)) {
-      filename = path.resolve(filename);
-      var stats = fs.lstatSync(filename);
-      if (stats.isDirectory()) {
-        filename = path.join(filename, '*.js');
+  function load (type, globPaths) {
+    var base = type === 'model' ? baseModel : baseCollection;
+    var gFiles = [];
+    for (var i = globPaths.length - 1; i >= 0; i--) {
+      var filename = globPaths[i];
+      if (!glob.hasMagic(filename)) {
+        filename = path.resolve(filename);
+        var stats = fs.lstatSync(filename);
+        if (stats.isDirectory()) {
+          filename = path.join(filename, '*.js');
+        }
       }
+      gFiles = glob.sync(filename);
+      modelObjects = modelObjects.concat(gFiles);
```

Comment:
It looks like `modelObjects` is defined at the top (line 9 of this file), but since it's referred to in this function, it's persisting its value for both models and collections. You should move the definition from line 9 to line 65, so that it starts with a new array each time `load` is called.